core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -2396,6 +2396,7 @@ wasm_call_function(WASMExecEnv *exec_env, WASMFunctionInstance *function,
 {
     WASMModuleInstance *module_inst =
         (WASMModuleInstance *)exec_env->module_inst;
+    module_inst->e->common.exec_env = exec_env;
```

Comment:
Had better add an empty line above or put it after `wasm_exec_env_set_thread_info(exec_env)`